File: frontend/src/admin/dto/CompletionMailReportDto.ts
export interface CompletionMailReportDto {
  totalUsers: number;
  successful: CompletionMailReportUserDto[];
  failed: Map<CompletionMailReportUserDto, string>;
}

interface CompletionMailReportUserDto {
  username: string;
  fullName: string;
  email: string;
}

File: backend/src/main/java/net/dnadas/training_portal/dto/user/CompletionMailReportDto.java
package net.dnadas.training_portal.dto.user;

import java.util.List;
import java.util.Map;

public record CompletionMailReportDto(
  Integer totalUsers,
  List<CompletionMailUserInternalDto> successful,
  Map<CompletionMailUserInternalDto, String> failed) {
}
